plugin/handlergen.go

```diff
@@ -470,13 +477,15 @@ func (p *OrmPlugin) generateListHandler(message *generator.Descriptor) {
 	p.P(`if err != nil {`)
 	p.P(`return nil, err`)
 	p.P(`}`)
+	if getMessageOptions(message).GetMultiAccount() {
+		p.generateAccoundIdWhereClause()
+	}
 	p.generateBeforeListHookCall(ormable, "ApplyQuery")
 	p.P(`db, err = `, p.Import(tkgormImport), `.ApplyCollectionOperators(ctx, db, &`, ormable.Name, `{}, &`, typeName, `{}, `, f, `,`, s, `,`, pg, `,`, fs, `)`)
 	p.P(`if err != nil {`)
 	p.P(`return nil, err`)
 	p.P(`}`)
 	p.generateBeforeListHookCall(ormable, "Find")
-	p.P(`db = db.Where(&ormObj)`)
```

Comment:
Is there a compelling reason to remove this? I know that using the &ormObj from ToOrm() can cause issues when ToOrm() initializes a child object, but skipping it like this seems like it could also have consequences.

Reply:
Initially this line was added here in order to apply AccountId condition to the List query clause. I decided to change it to a more clear way with `p.generateAccoundIdWhereClause()`.
But now I'm recalling that that was used by apps for applying some additional(non AccountId) filters(e.g. zone type). Is that true? If so, then I'll revert it.

Reply:
This is a two sided issue. Using `db.Where(&ormObj)` allows for filtering on fields filled set in the ToOrm hooks, but its use here (and possibly elsewhere) also restricts ToOrm in an unexpected way. Setting a synthetic object in ToOrm() can cause these queries to fail (I don't have the exact error handy), because you can't filter on an object that has associations.

I'd say the new approach is the better one, but I think it will also cause some issues when upgrading to this code (potentially hard to detect issues). 
If we want to do this more 'breaking' change, can we cut a release of gentool with the fix in atlas-gen-validate before including this in the next version? @Evgeniy-L